WordPress/Classes/Utility/Media/MediaImageExporter.swift

```diff
@@ -284,58 +284,48 @@ class MediaImageExporter: MediaExporter {
-                let thumbnailOptions: [NSString: Any] = [kCGImageSourceThumbnailMaxPixelSize: maximumSize,
-                                                       kCGImageSourceCreateThumbnailFromImageAlways: true,
-                                                       kCGImageSourceShouldCache: false,
-                                                       kCGImageSourceTypeIdentifierHint: sourceUTType,
-                                                       kCGImageSourceCreateThumbnailWithTransform: true]
-                // Create a thumbnail of the image source.
-                guard let image = CGImageSourceCreateThumbnailAtIndex(source, 0, thumbnailOptions as CFDictionary) else {
-                    throw ImageExportError.imageSourceThumbnailGenerationFailed
-                }
-
-                if nullifyGPSData == true {
-                    // When removing GPS data for a thumbnail, we have to remove the dictionary
-                    // itself for the CGImageDestinationAddImage method.
-                    imageProperties.removeValue(forKey: kCGImagePropertyGPSDictionary)
-                }
-
-                // Add the thumbnail image as the destination's image.
-                CGImageDestinationAddImage(destination, image, imageProperties as CFDictionary?)
-
-                // Get the dimensions from the CGImage itself
-                width = CGFloat(image.width)
-                height = CGFloat(image.height)
-            } else {
-
-                if nullifyGPSData == true {
-                    // When removing GPS data for a full-sized image, we have to nullify the GPS dictionary
-                    // for the CGImageDestinationAddImageFromSource method.
-                    imageProperties[kCGImagePropertyGPSDictionary] = kCFNull
-                }
-                // No resizing needed, add the full sized image from the source
-                CGImageDestinationAddImageFromSource(destination, source, 0, imageProperties as CFDictionary?)
-
-                // Get the dimensions of the full size image from the source's properties
-                width = imageProperties[kCGImagePropertyPixelWidth] as? CGFloat
-                height = imageProperties[kCGImagePropertyPixelHeight] as? CGFloat
+                thumbnailOptions[kCGImageSourceThumbnailMaxPixelSize] = maximumSize
+            }
+
+            // Create a thumbnail of the image source.
+            guard let image = CGImageSourceCreateThumbnailAtIndex(source, 0, thumbnailOptions as CFDictionary) else {
+                throw ImageExportError.imageSourceThumbnailGenerationFailed
+            }
+
+            if nullifyGPSData == true {
+                // When removing GPS data for a thumbnail, we have to remove the dictionary
+                // itself for the CGImageDestinationAddImage method.
+                imageProperties.removeValue(forKey: kCGImagePropertyGPSDictionary)
```

Comment:
I'm a bit uncertain here because we were previously removing the GPS data differently depending on whether we were dealing with a full sized image or a thumbnail. Do we still need the alternative code path too?

Reply:
We are removing it from the properties so the new image should have any GPS info.